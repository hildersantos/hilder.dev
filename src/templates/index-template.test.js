import React from "react"
import { render } from "react-testing-library"

import IndexTemplate from "./index-template"
import fixtureSiteInfo from "../../tests/__fixtures__/fixture-site-info"
import fixtureAllMarkdownRemark from "../../tests/__fixtures__/fixture-all-markdown-remark"
import { useStaticQuery, StaticQuery } from "gatsby"

const props = {
  ...fixtureSiteInfo,
  ...fixtureAllMarkdownRemark,
}

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) => render(props)),
    useStaticQuery.mockReturnValue(props)
})

describe("IndexTemplate", () => {
  it("renders without errors", () => {
    const { asFragment } = render(<IndexTemplate data={props} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("has a header", () => {
    const { container } = render(<IndexTemplate data={props} />)
    expect(container.querySelector("header")).toBeInTheDocument()
  })
})
