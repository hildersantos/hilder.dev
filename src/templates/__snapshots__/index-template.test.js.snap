// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexTemplate renders without errors 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <div
      class="container"
    >
      <header
        class="header-inner"
      >
        <a
          class="header-inner__link--home header-inner__link"
          href="/"
        >
          Test Title
        </a>
      </header>
      <section
        class="post-listing"
      >
        <article
          class="post-listing__article"
        >
          <h2
            class="post-listing__title"
          >
            <a
              class="post-listing__link"
              href="test_0"
            >
              Test 0
            </a>
          </h2>
        </article>
        <article
          class="post-listing__article"
        >
          <h2
            class="post-listing__title"
          >
            <a
              class="post-listing__link"
              href="test_1"
            >
              Test 1
            </a>
          </h2>
        </article>
        <article
          class="post-listing__article"
        >
          <h2
            class="post-listing__title"
          >
            <a
              class="post-listing__link"
              href="test_2"
            >
              Test 2
            </a>
          </h2>
        </article>
        <article
          class="post-listing__article"
        >
          <h2
            class="post-listing__title"
          >
            <a
              class="post-listing__link"
              href="test_3"
            >
              Test 3
            </a>
          </h2>
        </article>
      </section>
      <div
        class="navigation"
      >
        <ul
          class="navigation__list"
        >
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="test_0"
            >
              Test 0
            </a>
          </li>
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="test_1"
            >
              Test 1
            </a>
          </li>
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="test_2"
            >
              Test 2
            </a>
          </li>
          <li
            class="navigation__item"
          >
            <a
              class="navigation__link"
              href="test_3"
            >
              Test 3
            </a>
          </li>
        </ul>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
